### PR TITLE
Update dependencies and fix ESLint warnings accordingly

### PR DESCRIPTION
**Pull request by igor-savin-ht**

### CODE REVIEW

**Comment by igor-savin-ht** [comment]
@elhigu Would be great if you could take a look at https://github.com/tgriesser/knex/pull/2433/commits/aaa4d71e914faa1c4c1313c26bbdd26067b0b448 -> I've removed it as ESLint demanded me to do so, but I wonder if there was actually a bug there and that param was supposed to be used.

**Comment by igor-savin-ht** [comment]
@elhigu I can restore removed code with eslint ignore and add the ToDo to review it later, if it's not an option to check it now :)

**Comment by elhigu** [comment]
@igor-savin-ht honestly I just haven't got time to read this through with thought. Smaller changes are easier to accept with confidence that they are not breaking anything :) I would prefer eslint ignore for now.

**Comment by igor-savin-ht** [comment]
@elhigu Done!

**Comment by elhigu** [comment]
There were few regex where some escape backslashes were dropped. Are you sure they work exactly the same after the changes?

**Comment by igor-savin-ht** [comment]
@elhigu ESLint was complaining about those escape backslashes being redundant, I tend to trust its judgement on that. Here is an in-depth discussion on this subject: https://github.com/eslint/eslint/issues/7656
As is with unused variable, I'm open to replacing this change with either line-specific ignore or global rule to ignore useless escapes.

**Comment by elhigu** [comment]
I'd rather have that global rule set so I need to verify manually e.g. with runkit that each of those regexps are working as expected after change. In this case I don't trust knex testsuite enough and I believe that eslint has them correctly fixed, but I need to be sure.

File: src/query/string.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ function convertTimezone(tz) {
   if (tz === 'Z') {
     return 0;
   }
-  const m = tz.match(/([\+\-\s])(\d\d):?(\d\d)?/);
+  const m = tz.match(/([+\-\s])(\d\d):?(\d\d)?/);
```
Comment (elhigu) [inline src/query/string.js]: Actually I wonder why eslint left `\-`escaping there.

**Comment by elhigu** [comment]
All good https://runkit.com/embed/ygveesp9rad2

**Pull request merged**